sdk/cosmos/src/operations/list_databases.rs

```diff
@@ -0,0 +1,127 @@
+use crate::headers::from_headers::*;
+use crate::prelude::*;
+use crate::resources::Database;
+use crate::ResourceQuota;
+
+use azure_core::headers::{continuation_token_from_headers_optional, session_token_from_headers};
+use azure_core::{collect_pinned_stream, prelude::*, Request, Response};
+use chrono::{DateTime, Utc};
+
+#[derive(Debug, Clone)]
+pub struct ListDatabasesOptions {
+    consistency_level: Option<ConsistencyLevel>,
+    max_item_count: MaxItemCount,
+}
+
+impl ListDatabasesOptions {
+    pub fn new() -> Self {
+        Self {
+            consistency_level: None,
+            max_item_count: MaxItemCount::new(-1),
+        }
+    }
+
+    setters! {
+        consistency_level: ConsistencyLevel => Some(consistency_level),
+        max_item_count: i32 => MaxItemCount::new(max_item_count),
+    }
+
+    pub async fn decorate_request(&self, request: &mut Request) -> Result<(), crate::Error> {
+        azure_core::headers::add_optional_header2(&self.consistency_level, request)?;
+        azure_core::headers::add_mandatory_header2(&self.max_item_count, request)?;
+        Ok(())
+    }
+
+    // pub fn stream(&self) -> impl Stream<Item = Result<ListDatabasesResponse, crate::Error>> + '_ {
```

Comment:
We'll have to figure out what our interface for kicking off an async iterator response will be.

Reply:
:100: yess, will do as a follow-up!